manga_notify/drivers/basic_rss.py

```diff
@@ -0,0 +1,63 @@
+import typing
+import aiohttp
+
+import xml.etree.ElementTree as ET
+
+from . import driver
+from . import common_message
+from ..channels import channel
+from ..database import feed_storage
+
+
+UA = (
+    'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:93.0) '
+    'Gecko/20100101 Firefox/93.0'
+)
+
+
+def iter_items(items):
+    for item in items:
+        title = item.find('title').text
+        link = item.find('link').text
+        if not title:
+            continue
+        yield common_message.ParsingItem(
+            name=title,
+            link=link,
+        )
+
+
+class BasicRss(driver.Driver):
+    def is_match(self, url: str) -> bool:
+        return NotImplementedError
+
+    async def parse(
+        self,
+        feed_data: feed_storage.FeedData,
+    ) -> driver.ParsingResult:
+        headers = {
+            'User-Agent': UA,
+        }
+
+        async with aiohttp.ClientSession() as client:
+            async with client.get(
+                feed_data.get_url(),
+                headers=headers,
+            ) as response:
+                data = await response.text()
+
+        root = ET.fromstring(data)
+        parsed_items = []
+        for item in iter_items(root.findall('./channel/item')):
```

Comment:
Does readmanga have exactly the same rss feed format as mangasee?

Reply:
It seems like the only difference is guid permalink (which is unused)

```xml
<rss version="2.0">
  <channel>
    <atom:link href="https://mangasee123.com/rss/Destroy-All-Humankind-They-Cant-Be-Regenerated.xml" rel="self" type="application/rss+xml"/>
    <title>Destroy All Humankind. They Can't Be Regenerated.</title>
    <link>
      https://mangasee123.com/manga/Destroy-All-Humankind-They-Cant-Be-Regenerated
    </link>
    <lastBuildDate>Sat, 15 Oct 2022 14:34:10 +0000</lastBuildDate>
    <description>
      Destroy All Humankind. They Can't Be Regenerated. Latest Chapters
    </description>
    <language>en-us</language>
    <category>entertainment</category>
    <image>
      <url>
        https://temp.compsci88.com/cover/Destroy-All-Humankind-They-Cant-Be-Regenerated.jpg
      </url>
      <title>Destroy All Humankind. They Can't Be Regenerated.</title>
      <link>
        https://mangasee123.com/manga/Destroy-All-Humankind-They-Cant-Be-Regenerated
      </link>
    </image>
    <item>
      <title>
        Destroy All Humankind. They Can't Be Regenerated. Chapter 40
      </title>
      <link>
        https://mangasee123.com/read-online/Destroy-All-Humankind-They-Cant-Be-Regenerated-chapter-40-page-1.html
      </link>
      <pubDate>Sat, 15 Oct 2022 15:04:42 +0000</pubDate>
      <guid isPermaLink="false">Destroy-All-Humankind-They-Cant-Be-Regenerated-40</guid>
    </item>
  </channel>
</rss>
```
vs
```xml
<rss version="2.0">
  <channel>
    <atom:link href="https://readmanga.live/rss/manga?name=pesn_nochnyh_sov" rel="self" type="application/rss+xml"/>
    <title>Манга Песнь ночных сов</title>
    <description>
      Последние глав и обновления для манги Песнь ночных сов Song of the Night Walkers
    </description>
    <link>https://readmanga.live/pesn_nochnyh_sov__A35c96</link>
    <image>
      <url>
        https://staticrm.rmr.rocks/uploads/pics/01/43/077.jpg
      </url>
      <title>Манга Песнь ночных сов</title>
      <link>https://readmanga.live/pesn_nochnyh_sov__A35c96</link>
    </image>
    <lastBuildDate>Mon, 03 Oct 2022 07:09:47 +0000</lastBuildDate>
    <item>
      <title>Песнь ночных сов: 9 - 80 Ты у Меня Первая, Семпай.</title>
      <link>
        https://readmanga.live/pesn_nochnyh_sov__A35c96/vol9/80
      </link>
      <guid isPermaLink="true">
        https://readmanga.live/pesn_nochnyh_sov__A35c96/vol9/80
      </guid>
      <pubDate>Sat, 15 Oct 2022 13:51:14 +0000</pubDate>
      <author>L4st_ (Переводчик)</author>
      <description>
        Песнь ночных сов Song of the Night Walkers: 9 - 80 Ты у Меня Первая, Семпай.
      </description>
    </item>
  </channel>
</rss>
```

(full: [readmanga rss](https://readmanga.live/rss/manga?name=pesn_nochnyh_sov) vs [mangasee123](https://mangasee123.com/rss/Destroy-All-Humankind-They-Cant-Be-Regenerated.xml))